src/Name/Generator/NameGeneratorChain.php

```diff
@@ -23,8 +23,8 @@ public function generate(): string
     {
         $queue = clone $this->queue;
         while (false === $queue->isEmpty()) {
-            if ('' !== ($debugId = $queue->extract()->generate())) {
-                return $debugId;
+            if ('' !== ($name = $queue->extract()->generate())) {
```

Comment:
Is it special behavior for name generators - if them return an empty string, then caller may think it is like a 'I cannot generate name, try another one'?